File: 30 days of code/__ Java/Day 21 - Generics.java
/**
 *              GENERICS | JAVA
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 21
 */

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws FileNotFoundException {
        File file = new File("input.txt");
        Scanner scan = new Scanner(file);
        {
            Integer[] intArray = new Integer[scan.nextInt()];
            for (int i = 0; i < intArray.length; ++i) {
                intArray[i] = scan.nextInt();
            }

            String[] stringArray = new String[scan.nextInt()];
            for (int i = 0; i < stringArray.length; ++i) {
                stringArray[i] = scan.next();
            }

            Printer<Integer> intPrinter = new Printer<>();
            Printer<String> stringPrinter = new Printer<>();

            intPrinter.printArray(intArray);
            stringPrinter.printArray(stringArray);
        }
        scan.close();
    }
}

class Printer <T> {
    void printArray(T[] array) {
        for (T t : array) {
            System.out.println(t);
        }
    }
}

File: 30 days of code/__ C++/Day 21 - Generics.cpp
/**
*              GENERICS | C++
* Domain       30 DAYS OF CODE
* Sub domain   DAY 21
*/

#include "stdafx.h"
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

using namespace std;

ifstream file("input.txt");
istream& f = file;

template<typename T>
void printArray(vector<T> array) {
	for (int i = 0; i < array.size(); ++i) {
		cout << array[i] << endl;
	}
}

void main() {
	int n; f >> n;
	vector<int> intArray(n);
	for (int i = 0; i < n; ++i) {
		f >> intArray[i];
	}

	f >> n;
	vector<string> stringArray(n);
	for (int i = 0; i < n; ++i) {
		f >> stringArray[i];
	}

	printArray<int>(intArray);
	printArray<string>(stringArray);
}